src/service.rs

```diff
@@ -23,13 +30,162 @@ impl Service for ApplicationService {
     type Parameters = ();
 
     async fn new(runtime: ServiceRuntime<Self>) -> Self {
-        let state = Application::load(runtime.root_view_storage_context())
-            .await
-            .expect("Failed to load state");
-        ApplicationService { state, runtime }
+        ApplicationService {
+            runtime: Arc::new(Mutex::new(runtime)),
+        }
+    }
+
+    async fn handle_query(&self, query: Self::Query) -> Self::QueryResponse {
+        Schema::build(
+            EmptyFields,
+            Mutation {
+                runtime: self.runtime.clone(),
+            },
+            EmptySubscription,
+        )
+        .finish()
+        .execute(query)
+        .await
+    }
+}
+
+/// Root type that defines all the GraphQL mutations available from the service.
+pub struct Mutation {
+    runtime: Arc<Mutex<ServiceRuntime<ApplicationService>>>,
+}
+
+#[async_graphql::Object]
+impl Mutation {
+    /// Executes a chat completion using the Atoma Network.
+    async fn chat(
+        &self,
+        api_token: String,
+        message: ChatMessage,
+    ) -> async_graphql::Result<Vec<u8>> {
+        let request = ChatCompletionRequest {
+            stream: false,
+            messages: &[&message],
+            model: "meta-llama/Llama-3.3-70B-Instruct".to_owned(),
+            max_tokens: 128,
```

Comment:
I suggest these values are also input arguments

Reply:
Done, but kept the default values in case they're not specified in the GraphQL request.